### PR TITLE
FreeBSD ci

### PR DESCRIPTION
**Pull request by TingDaoK**
- experiments about FreeBSD

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -186,6 +186,28 @@ jobs:
...
+        with:
+          submodules: true
+      - name: Build ${{ env.PACKAGE_NAME }} + consumers
+        uses: vmactions/freebsd-vm@v0
+        with:
+          envs: 'AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_DEFAULT_REGION AWS_REGION'
+          usesh: true
+          sync: rsync
+          copyback: false
+          run: |
```
Comment (DmitriyMusatkin) [inline .github/workflows/ci.yml:204]: nit: this action has a prepare step, which might be a better place for installing stuff like pip

File: test/test_checksums.py [hunk 2]
```diff
@@ -80,11 +81,13 @@ def test_crc32c_large_buffer(self):
         self.assertEqual(0xfb5b991d, val)
 
     def test_crc32c_huge_buffer(self):
-        # stress the internal logic that handles buffers larger than C's INT_MAX
+        if sys.platform.startswith('freebsd'):
```
Comment (DmitriyMusatkin) [inline test/test_checksums.py:84]: only fails for crc32c? do we want to skip other flex checksum cases for huge buffer?

**Review by DmitriyMusatkin — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -7,8 +7,8 @@ on:
       - 'docs'
 
 env:
-  BUILDER_VERSION: v0.9.44
-  BUILDER_SOURCE: releases
+  BUILDER_VERSION: freebsd-ci
```
Comment (DmitriyMusatkin) [inline .github/workflows/ci.yml:10]: dont forget to revert before pushing

File: test/test_checksums.py (hunk 2 above)
Comment (TingDaoK) [reply test/test_checksums.py:87]: It only fails for crc32c for now, I skip the other huge_buffer test for safe.

**Pull request merged**